wazo_websocketd/auth.py

```diff
@@ -99,7 +98,9 @@ def _calculate_next_check(self, now, expires_at):
 class Authenticator:
     def __init__(self, config):
         self._async_auth_client = AsyncAuthClient(config)
-        auth_check_class = STRATEGIES.get(config['auth_check_strategy'])
+        auth_check_class: Callable | None = STRATEGIES.get(
```

Comment:
Is this a callable? Looking at the file, it seems to expect a class. Ideally the two classes used should  extend some kind of abtract class so the definition is uniform and we could use that. If not, you could create a `Protocol` to type the expected signature. Example in this case it could be:

```python
class AuthChecker(Protocol):
    def __init__(self, async_auth_client: AsyncAuthClient, config: ConfigDict) -> None:
        ...

    async def run(self, token_getter: Callable[[], TokenDict]) -> None:
        ...
```

Reply:
I'll go with abstract class. It should be used from the beginning to know which method to implement